polaris/dataset/_dataset.py

```diff
@@ -192,6 +193,11 @@ def _compute_checksum(table):
         table_hash = pd.util.hash_pandas_object(df, index=False).sum()
         hash_fn.update(table_hash)
 
+        # If the Zarr archive exists, we hash its contents too.
+        if zarr_root_path is not None:
+            zarr_hash = compute_zarr_checksum(zarr_root_path)
+            hash_fn.update(zarr_hash.encode())
```

Comment:
**Comment**
@cwognum and I had a live discussion about this yesterday. See my general review comments.